File: 0061-rotate-list/0061-rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if (head == null || head.next == null) {
            return head;
        }
        int l=1;
        ListNode tail=head;
        while(tail.next!=null){
            l++;
            tail=tail.next;
        }
        k=k%l;
        if(k==0){
            return head;
        }
        ListNode newt=head;
        for(int i=0;i<l-k-1;i++){
            newt=newt.next;
        }
        ListNode newh=newt.next;
        newt.next=null;
        tail.next=head;
        return newh;
    }

}

File: 0061-rotate-list/0061-rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if not head or not head.next:
            return head
        length=1
        temp=head
        while temp.next:
            length+=1
            temp=temp.next
        temp.next=head
        k=k%length
        if k==0:
            temp.next=None
            return head
        taillength=length-k
        newtail=head
        for _ in range(taillength-1):
            newtail=newtail.next
        newhead=newtail.next
        newtail.next=None
        return newhead